### PR TITLE
circleci: Push to app collection

### PR DESCRIPTION
**Pull request by tfussell**
Towards https://github.com/giantswarm/giantswarm/issues/8655 and https://github.com/giantswarm/giantswarm/issues/8436

This PR implements a CircleCI workflow for pushing releases as a helm check to the catalog and provider app collections so they can be automatically deployed to control planes.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,17 +1,75 @@
-version: 2
+
+version: 2.1
+
+orbs:
+  architect: giantswarm/architect@dev:persist-attach
```
Comment (tfussell) [inline .circleci/config.yml]: This will be updated once https://github.com/giantswarm/architect-orb/pull/82 is released.

Comment (MarcelMue) [reply .circleci/config.yml]: Is there anything blocking us from releasing the orb now? It should be merged now, right?

Comment (tfussell) [reply .circleci/config.yml]: I think so! I was just making sure I didn't make any mistakes here first. I'll tag the orb commit and get all of this merged.

Comment (tfussell) [reply .circleci/config.yml]: Done.

**Comment by tfussell** [comment]
The orb is released and updated here now. Here's the commit on the `control-plane-test-catalog`: https://github.com/giantswarm/control-plane-test-catalog/commit/897d215c7e44f6e0c949568bc5ef6cecb7079efc from the latest build. I installed the chart on `ginger` manually using `helm install https://giantswarm.github.io/control-plane-test-catalog/releases-kvm-0.0.0-a0e045ed25f471017f324f12d19322d6a45236d1.tgz --name releases-kvm-unique`. The output of `kubectl get releases` is as follows:
```
kubectl get releases
NAME      KUBERNETES VERSION   STATE        AGE
v11.0.0   1.16.3               deprecated   56d
v11.1.0   1.16.3               active       37d
v11.2.0   1.16.3               active       9d
v8.0.0    1.14.1               deprecated   351d
v8.1.0    1.14.1               deprecated   311d
v8.2.0    1.14.3               deprecated   256d
v8.2.1    1.14.4               deprecated   256d
v8.3.0    1.14.5               deprecated   199d
v8.4.0    1.14.6               deprecated   199d
v9.0.0    1.15.5               active       130d
```
On merge to `master` this will go out on all KVM control planes which I think is fine. Additionally, we will temporarily be unable to run `opsctl update draughtsman` for KVM until we make other changes to our deployment process to handle the slightly changed `kvm.yaml` format.

**Comment by MarcelMue** [comment]
```
Additionally, we will temporarily be unable to run opsctl update draughtsman for KVM until we make other changes to our deployment process to handle the slightly changed kvm.yaml format.
```

This is an issue IMO as this blocks releases temporarily. Why do you think we can't run `opsctl update draughtsman`? I thought it would be fine. Have you tried?

**Comment by tfussell** [comment]
> ```
> Additionally, we will temporarily be unable to run opsctl update draughtsman for KVM until we make other changes to our deployment process to handle the slightly changed kvm.yaml format.
> ```
> 
> This is an issue IMO as this blocks releases temporarily. Why do you think we can't run `opsctl update draughtsman`? I thought it would be fine. Have you tried?

I wanted to, for example, `kvm.yaml` directly to the chart `values.yaml`. Helm doesn't allow top level arrays for `values.yaml` so I had to change `kvm.yaml` by putting it all under a top-level `versions` key. This would make `opsctl update draughtsman` not work.

Since I wasn't happy with this, I used a little bash magic to indent `kvm.yaml` and put it under a top level key in `values.yaml`. This means the structure is unchanged and nothing should break. Win win!

**Comment by tfussell** [comment]
Closing in favor of https://github.com/giantswarm/releases/pull/148

**Pull request closed without merging**